Deployment/Jenkins/master/Master-Elbv1.groovy

```diff
@@ -0,0 +1,138 @@
+pipeline {
+
+agent any
+
+options {
+    buildDiscarder(
+        logRotator(
+            numToKeepStr: '5',
+            daysToKeepStr: '30',
+            artifactDaysToKeepStr: '30',
+            artifactNumToKeepStr: '3'
+        )
+    )
+    disableConcurrentBuilds()
+    timeout(time: 60, unit: 'MINUTES')
+}
+
+environment {
+    AWS_DEFAULT_REGION = "${AwsRegion}"
+    AWS_CA_BUNDLE = '/etc/pki/tls/certs/ca-bundle.crt'
+    REQUESTS_CA_BUNDLE = '/etc/pki/tls/certs/ca-bundle.crt'
+}
+
+parameters {
+    string(name: 'AwsRegion', defaultValue: 'us-east-1', description: 'Amazon region to deploy resources into')
+    string(name: 'AwsCred', description: 'Jenkins-stored AWS credential with which to execute cloud-layer commands')
+    string(name: 'GitCred', description: 'Jenkins-stored Git credential with which to execute git commands')
+    string(name: 'GitProjUrl', description: 'SSH URL from which to download the Jenkins git project')
+    string(name: 'GitProjBranch', description: 'Project-branch to use from the Jenkins git project')
+    string(name: 'CfnStackRoot', description: 'Unique token to prepend to all stack-element names')
+    string(name: 'TemplateUrl', description: 'S3-hosted URL for the EC2 template file')
+    string(name: 'BackendTimeout', defaultValue: '600', description: 'How long - in seconds - back-end connection may be idle before attempting session-cleanup')
+    string(name: 'JenkinsAgentPort', description: 'TCP Port number that the Jenkins agent-hosts connect through')
+    string(name: 'JenkinsListenerCert', defaultValue: '', description: 'Name/ID of the ACM-managed SSL Certificate to protect public listener')
+    string(name: 'JenkinsListenPort', defaultValue: '443', description: 'TCP Port number on which the Jenkins ELB listens for requests')
+    string(name: 'JenkinsServicePort', defaultValue: '8080', description: 'TCP Port number that the Jenkins host listens to')
+    string(name: 'JenkinsPassesSsh', defaultValue: 'false', description: 'Whether to allow SSH passthrough to Jenkins master')
+    string(name: 'HaSubnets', description: 'Select three subnets - each from different Availability Zones')
+    string(name: 'ProxyPrettyName', description: 'A short, human-friendly label to assign to the ELB - no capital letters')
+    string(name: 'SecurityGroupIds', description: 'List of security groups to apply to the ELB')
+
+}
+
+stages {
+    stage ('Prepare Instance Environment') {
+        steps {
+            deleteDir()
+            git branch: "${GitProjBranch}",
+                credentialsId: "${GitCred}",
+                url: "${GitProjUrl}"
+            writeFile file: 'master.ec2.instance.parms.json',
+                text: /
+                [
```

Comment:
Readability nit-pick: probably want to fix the indentation. All text between the `text: /` and closing `/` should be indented.

Reply:
gave it a shot